src/resolverHelpers.js

```diff
@@ -1,60 +1,77 @@
 const request = require('request-promise');
 
-const { graphRoot }  = require('./constants');
+const { graphRoot } = require('./constants');
 
-function makeRequest(path, authorization, isSecondary) {
+function makeRequest(path, authorization, args, isSecondary) {
+  if (args.id != null) {
+    url = url + '/' + args.id;
+  }
   var options = {
-      url: path,
-      headers: {
-          'Authorization': 'bearer ' + authorization
-      }
+    url: path,
+    headers: {
+      'Authorization': 'bearer ' + authorization
+    }
   };
-  return request(options).then(function (requestBody, error) {
-      if (!error) {
-          let body = JSON.parse(requestBody);
-          if (Object.keys(body).includes('value')) {
-              var next = body['value'];
-              if (Array.isArray(next)) {
-                  next.map(value => value['__path'] = path + '/' + value['id']);
-                  next.map(value => value['__secondary'] = true);
-                  next.map(value => value['__session'] =  authorization);
-              } else {
-                  next['__path'] = path;
-                  next['__secondary'] = true;
-                  next['__session'] = authorization;
-              }
-              return next;
-          } else {
-              body['__path'] = path;
-              body['__session'] = authorization;
-              if (isSecondary) {
-                  body['__secondary'] = true;
-              }
-              return body;
-          }
+  return request(options).then(function (responseBody, error) {
+    if (!error) {
+      let body = JSON.parse(responseBody);
+      if (Object.keys(body).includes('value')) {
+        var next = body['value'];
+        if (Array.isArray(next)) {
+          next.map(value => value['__path'] = path + '/' + value['id']);
+          next.map(value => value['__secondary'] = true);
+          next.map(value => value['__session'] = authorization);
+        } else {
+          next['__path'] = path;
+          next['__secondary'] = true;
+          next['__session'] = authorization;
+        }
+        return next;
       } else {
-          console.log(error);
-          return 'error';
+        body['__session'] = authorization;
+        if (isSecondary) {
+          body['__secondary'] = true;
+        }
+        if (args.id != null) {
+          body['__path'] = path + '/' + args.id;
+          if (!Array.isArray(body)) {
+            body = [body];
+          }
+        } else {
+          body['__path'] = path;
+        }
+        return body;
       }
+    } else {
+      console.log(error);
+      return 'error';
+    }
+  }).catch(function (err) {
+    return {}; // return empty object 
```

Comment:
Maybe add a `console.error` here or something it doesn't fail silently?

Reply:
Yes, thanks for the suggestion.